### PR TITLE
feat: add `widgetbook_addon` package

### PR DESCRIPTION
**Pull request by jenshor**
- adds `widgetbook_addon` package

### CODE REVIEW

**Review by YoussefRaafatNasry — APPROVED** [review]

File: packages/widgetbook_addon/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## Overview
+
+A placeholder package for Widgetbook's Add-on system.
```
Comment (jenshor) [inline packages/widgetbook_addon/README.md:3]: Need to improve this later

File: packages/widgetbook_addon/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+## 3.0.0-beta.0
+
+- Initial version.
```
Comment (jenshor) [inline packages/widgetbook_addon/CHANGELOG.md:3]: Need to improve this later, but might be covered by melos changelog generation anyways

File: packages/widgetbook_addon/lib/src/widgetbook_addon.dart [hunk 3]
```diff
@@ -37,6 +25,9 @@ abstract class WidgetbookAddOn<T extends WidgetbookAddOnModel> {
   final T setting;
   late ValueNotifier<T> provider;
 
+  /// Updates the router's query parameters using the changed [value].
+  void updateQueryParameters(BuildContext context, T value);
+
```
Comment (jenshor) [inline packages/widgetbook_addon/lib/src/widgetbook_addon.dart:30]: Really like this refactor 👍🏻

**Review by jenshor — COMMENTED** [review]
It's a bit weird that I'm reviewing my own PR, but good job, @YoussefRaafatNasry 🎉

**Pull request merged**